benchmark/run_all_benchmarks.sh

```diff
@@ -273,120 +370,180 @@ run_preconditioner_benchmarks() {
 
 
 ################################################################################
-# SuiteSparse collection
-
-SSGET=ssget
-NUM_PROBLEMS="$(${SSGET} -n)"
+# Batch matrix functionality
 
-# Creates an input file for $1-th problem in the SuiteSparse collection
-generate_suite_sparse_input() {
-    INPUT=$(${SSGET} -i "$1" -e)
-    cat << EOT
-[{
-    "filename": "${INPUT}",
-    "problem": $(${SSGET} -i "$1" -j)
-}]
-EOT
+# Because ls -la always shows first 3 non-essential (for our purposes) lines
+count_num_batch_entries() {
+    echo $(($(ls -la $1 | wc -l) - 3))
 }
 
-parse_matrix_list() {
+# Read the list of the batch_mat file and prepare the batch matrix class list.
+parse_batch_matrix_list() {
     local source_list_file=$1
     local benchmark_list=""
-    local id=0
     for mtx in $(cat ${source_list_file}); do
-        if [[ ! "$mtx" =~ ^[0-9]+$ ]]; then
-            if [[ "$mtx" =~ ^[a-zA-Z0-9_-]+$ ]]; then
-                id=$(${SSGET} -s "[ @name == $mtx ]")
-            elif [[ "$mtx" =~ ^([a-zA-Z0-9_-]+)\/([a-zA-Z0-9_-]+)$ ]]; then
-                local group="${BASH_REMATCH[1]}"
-                local name="${BASH_REMATCH[2]}"
-                id=$(${SSGET} -s "[ @name == $name ] && [ @group == $group ]")
-            else
-                >&2 echo -e "Could not recognize entry $mtx."
-            fi
-        else
-            id=$mtx
-        fi
-        benchmark_list="$benchmark_list $id"
+        benchmark_list="$benchmark_list $mtx"
     done
     echo "$benchmark_list"
 }
 
-if [ $use_matrix_list_file -eq 1 ]; then
-    MATRIX_LIST=($(parse_matrix_list $MATRIX_LIST_FILE))
-    NUM_PROBLEMS=${#MATRIX_LIST[@]}
+if [ $use_batch_matrix -eq 1 ]; then
+    BATCH_MATRIX_LIST=$(parse_batch_matrix_list $BATCH_MATRIX_LIST_FILE)
```

Comment:
I guess this is the list of matrix classes? In that case, it would be easier to understand with
```suggestion
    BATCH_MATRIX_TYPES_LIST=$(parse_batch_matrix_list $BATCH_MATRIX_LIST_FILE)
```